src/utils/EventEmitter.js

```diff
@@ -64,14 +69,18 @@ export default class EventEmitter {
    * @param {EventHandler} eventHandler
    */
   removeListener(eventName, eventHandler) {
-    this._getEventListByName(eventName).delete(eventHandler)
+    const currentHandlers = this._getHandlersByEventName(eventName)
+    const index = currentHandlers.indexOf(eventHandler)
+    if (index > -1) {
+      currentHandlers.splice(index, 1)
+    }
   }
 
   /**
    * @param {string} eventName
    */
   removeAllListeners(eventName) {
-    this._getEventListByName(eventName).clear()
+    this.events[eventName] = []
```

Comment:
A slightly more performant way to empty an array is to set its `length` to `0`.

Moreover, IIUC, this `removeAllListeners()` might be called by users with an `eventName` that has never been registered before, in which case the value here would be undefined (instead of an `[]`, or `Set` before). A simple `if` guard protects that from throwing an exception:

```js
if (this.events[eventName] != null) this.events[eventName].length = 0
```

Reply:
What if we just `delete this.events[eventName]` here? 

- probably even better in terms of memory usage?
- slightly more readable than `= []` and `.length = 0`

Reply:
https://github.com/sweetalert2/sweetalert2/pull/2763/commits/de4f0743cea742d9f36d04fc10500c720143243c

Reply:
I think it's a reasonable solution.

----

Since we're talking about it, and you've said you were interested in the learning/feedback...

*Technically*, I think this is a little less memory/perf efficient.

1. mutating the object structure (`this.events`) by deleting the property causes a bit of de-optimization in the internals of the JS engine, as it has to keep discarding its "internal class" that represents the structure.

    You can't really avoid that on the event-registration side, unless you were to pre-populate the `this.events` object with all the known event name properties -- in your case, this is probably totally safe and OK since it's not a *general* event-emitter... you have a small fixed set of event names you support and fire, and you're not trying to support a use-case where users fire their own arbitrary event names. So you *could* just do `this.events = { didOpen: [], ... }` to preset the fixed object structure, which is *the most optimal way* from a JS engine internals perspective.
    
2. It's also susceptible to unnecessarily re-creating new arrays if there's re-registration of events after listeners were removed.

    For example, imagine a scenario where someone is doing "one"-style event handling, where it registers, fires, unregisters... and they repeat that with every single time they open the modal... over time there's a churn of creating and throwing away arrays, which leads to more garbage to GC. Minor amounts, sure, but still a tiny bit wasteful.

The `.length = 0` is not only faster (at least as fast as `delete`) but it preserves the allocation of the array object itself, for future re-use, so there's less GC over time. Of course, the `length = 0` requires an extra `if`, so is there a fraction of a microsecond cost there? Yeah.

It's a judgement call, honestly. If this were a big deal, I'd be pressing more for it. I'm just helping you understand the things that go through my head when I'm doing such tasks.

Reply:
One last *design decision* kind of feedback to consider:

*I prefer* if the various paths where events can be removed all end up in the same empty/bottom state. That is, if you remove listeners one at a time, you end up with an empty array (via `splice()`, right?). But if you remove them all at once, here with `delete` you'll end up not with an empty array but an `undefined` property.

There's a consistency concern here, but there's also the little bump you might trip over in the future, if you *forget* that it can be either empty array or `undefined`, and you introduce a bug.

For consistency and bug-risk-reduction sake, either I'd suggest keeping the `delete` *and* then also doing the same in the `off()` unregister handler once an array gets to empty... or getting rid of the `delete` and using the `length = 0` approach.

Reply:
And... if you really want to [take the *red pill* on GC optimization here](https://en.wikipedia.org/wiki/Red_pill_and_blue_pill)...

I'm [actually currently building an event-emitter library](https://github.com/byojs/eventer/blob/9220e4949afa9c00444fc9ef88095653d9cbc8ec/src/eventer.js#L28), and the specific design constraint driving it is to support *weak event listeners*. That's a very niche technique for hyper-optimizing a program by preventing blockers to GC of objects (and even function closures) caused by event listeners (functions) being held strongly (de facto, such as how you're doing it) by the event emitter. You accomplish "weak event listeners" with *quite advanced* techniques/features, including `WeakMap` (from back in ES6) + `WeakRef` / `FinalizationRegistry` (relatively new to JS in the last couple of years).

I roughly knew how the basics of how those features worked, but in tackling this task, now I've have been eyeballs deep in learning all their ins and outs over the last several days -- certainly part of the reason that "event handling" was on mind with respect to **SweetAlert**. :)

Reply:
> There's a consistency concern here, but there's also the little bump you might trip over in the future, if you forget that it can be either empty array or undefined, and you introduce a bug.

This was a very good point. Switched to `.length = 0` in https://github.com/sweetalert2/sweetalert2/pull/2763/commits/a84a42b12d8f3d229e596919a23e20c261befb16

Thanks!

Reply:
All LGTM.